packages/wonder-blocks-core/src/hooks/use-latest-ref.ts

```diff
@@ -10,7 +10,7 @@ import * as React from "react";
  * be passed to useEffect without causing unnecessary re-renders. See the
  * Storybook docs for a usage example.
  */
-export function useLatestRef<T>(value: T): React.RefObject<T> {
+export function useLatestRef<T>(value: T): React.MutableRefObject<T> {
```

Comment:
**question:** Do you want folks to be able to set current outside the ref? Should this be our own type it returns that meets your requirements without making the return value mutable?

Reply:
Yeah, I think using a custom type `{readonly current: T}` would be good. Note that `readonly` doesn't actually give you much protection because readonly types are assignable to mutable types! See: https://www.typescriptlang.org/play?#code/C4TwDgpgBAShCGATA9gOwDYjgMwPICMArCAY2AB4AVAPigF4oBvAJwRQxChIFdnXVgALiiUAvgChQkKAFluwePnQQcBYmSq0GjHnwgDhY8eOzdUZAJZooAW3gBrFWzSYcmgBS7+QkQEphcEguWBB4RKQUNEziULFQrMC8qExe+sASEiZmltZ2jnIKSk7YHqkGfsIFisqq4RpRjDFxCUkpvN4ZxiRoAM7AUPCV8tXFahHkqNw2+BDMWrYOTkEcOO4AjABMAMy+4t2ofVD4Ac4roWMak9Oz83kQVUWrmzviQA

Reply:
That isn't a fair example. After all, it's creating a new object in each case and `readonly` only governs the property it is declared on, not the value. You would need `as const` to cover the value too.

By creating a new object, you remove the readonly-ness.

Reply:
Is this a better example? https://www.typescriptlang.org/play?#code/C4TwDgpgBAShCGATA9gOwDYjgMwPICMArCAY2AB4AVAPigF4oBvAJwRQxChIFdnXVgALiiUAvgChQkKAFluwePnQQcBYmSq0GjHnwgDhY8eOzdUZAJZooAW3gBrFWzSYcmgBS7+QkQEphcEguWBB4RKQUNEziULFQrMC8qExe+sASEuLKwPFsuBwBzhyq4Rqo3Db4EMxatg5OQcWh7gCMAEwAzL7GAPQ9UAACwADOALQQAB6QZON8yMzifbLy8MAWqADmUPCwRa6hahFQFsNQqMg58OjoyADuEIgAdOKsSPmYj6kC9FAALACsADZev0ABJ3CAAN2qj1hi36IGQ3C48GS8GGwwsGzRu0a+zC6hywGQ2ygkPgzAsimUUCkD22pzkCiUTgJEVhz2ythWLOETOprMOZQqVRqP1eiHeIHh21QiFpAAt9FBEdwAOSsbbXCHymwrYDQCzAACE4j1zOUn143h+LSAA

Reply:
Yes, that's much better and really shows the issue. It's a shame that TypeScript doesn't handle better the assignment of readonly objects to mutable equivalents. I think even wrapping with `ReadOnly<>` doesn't help that. Probably need to exercise the muscle of using `Object.freeze` on things to enforce readonliness at runtime to help mitigate.